Move build dependencies into separate script for next

grische opened the pull request:
This PR is similar to https://github.com/freifunkMUC/site-ffm/pull/155 but with a different list of build dependencies.
The size of the docker container here is also ~420MB after the update and ~1GB before.
The differences between stable (red) and next (green) are:
```diff
diff --git a/install_build_dependencies.sh b/install_build_dependencies.sh
index 2d44437..879b853 100755
--- a/install_build_dependencies.sh
+++ b/install_build_dependencies.sh
@@ -21,6 +21,7 @@ apt-get -y --no-install-recommends install ca-certificates
 apt-get -y --no-install-recommends install \
     bash \
     bzip2 \
+    curl \
     diffutils \
     file \
     g++ \
@@ -31,7 +32,8 @@ apt-get -y --no-install-recommends install \
     make \
     patch \
     perl \
-    python2 \
+    python3 \
+    rsync \
     tar \
     unzip \
     wget
```

I verified that they are all required by running the build process in an `ubuntu:20.04` container, similar to the previous PR.

grische committed:
Move dependency installation to separate script

grische committed:
Reduce to mininmal amount of required packages

Removes all packages which are not required for the build process.
Most tools and libraries are being built by the OpenWRT build process
and do not need to be present on the CI builder.

goligo commented:
```
2021-10-28T20:26:40.4144937Z WARNING: Install qemu-img to create VDI/VMDK images
````

goligo commented:
Komisch - in stable hast Du das auch rausgeschmissen, da gab es aber keinen Buildfehler?

goligo commented:
Ah, haben sie tatsächlich rausgeworfen aus OpenWrt - muss also für next wieder rein.
https://github.com/openwrt/openwrt/commit/5876ba6460f14b4467812bdd76b9f3db323afb33

goligo committed:
Add qemu-utils again, which is required in next

OpenWrt builds for x86 also create VM-images. for this purpose qemu-img is used. While earlier releases built their own version to do so, starting with 21.02 it is required to be available on the build machine. See https://github.com/openwrt/openwrt/commit/5876ba6460f14b4467812bdd76b9f3db323afb33
